hedy.py

```diff
@@ -1310,29 +1301,26 @@ def turn(self, args):
         if arg.lstrip("-").isnumeric():
             return self.make_turn(arg)
 
-    def punctuation(self, args):
-        return ''.join([str(c) for c in args])
     def var(self, args):
         name = args[0]
         self.check_var_usage(args)
         return escape_var(name)
     def var_access(self, args):
         name = args[0]
         return escape_var(name)
-    def print(self, args):
-        argument_string = ""
-        i = 0
 
-        for argument in args:
-            # final argument and punctuation arguments do not have to be separated with a space, other do
-            if i == len(args)-1 or args[i+1] in self.punctuation_symbols:
-                space = ''
+    def print(self, args):
+        args_new = []
+        for a in args:
+            # list access has been already rewritten since it occurs lower in the tree
+            # therefore we should not split it anymore and thread it as 1 variable:
```

Comment:
This is a super minor thing, but I am not completely sure what this comment means. Do you mean that any list access would appear as a child node of the current print node, so by the time we reach this piece of code, the list access would already be transpiled, so we could treat it as a variable?

Reply:
Fair enough! I will make the explanation a bit more clear with an example!

Reply:
(but your interpretation is 100% correct!)